src/main/java/com/rapid7/client/dcerpc/msrrp/messages/BaseRegGetKeySecurityRequest.java

```diff
@@ -0,0 +1,94 @@
+
+<p><strong>SecurityInformation:</strong> The information that is
+needed to determine the type of security that is returned in <em>pRpcSecurityDescriptorOut</em>.
+See <a href="https://msdn.microsoft.com/en-us/library/cc244926.aspx">SECURITY_INFORMATION</a>
+(includes a list of possible <a href="https://msdn.microsoft.com/en-us/library/cc244893.aspx#gt_7bc13a0c-ca1a-4c78-b1bf-67e243f778bb">values</a>).</p>
+
+<p><strong>pRpcSecurityDescriptorIn: </strong>A pointer to a
+buffer containing a security descriptor. The client MUST provide a pointer to a
+<a href="https://msdn.microsoft.com/en-us/library/cc244924.aspx">RPC_SECURITY_DESCRIPTOR</a>
+with arbitrary contents. The server uses the size of this security descriptor
+to validate the client has the correct amount of memory allocated for the
+RPC_SECURITY_DESCRIPTOR pointed to by the <em>pRpcSecurityDescriptorOut</em>
+parameter</p>
+
+<p><strong>pRpcSecurityDescriptorOut:</strong> A pointer to a
+buffer to which the requested security descriptor MUST be written.</p>
+
+<p><strong>Return Values: </strong>The method returns 0
+(ERROR_SUCCESS) to indicate success; otherwise, it returns a nonzero error
+code, as specified in <a href="https://msdn.microsoft.com/en-us/library/cc231196.aspx">[MS-ERREF]</a>
+section <a href="https://msdn.microsoft.com/en-us/library/cc231199.aspx">2.2</a>.
+The most common error codes are listed in the following table.</p>
+@see <a href="https://msdn.microsoft.com/en-us/library/cc244936.aspx">https://msdn.microsoft.com/en-us/library/cc244936.aspx</a>
+ */
+public class BaseRegGetKeySecurityRequest extends RequestCall<BaseRegGetKeySecurityResponse> {
+    public static short OP_NUM = 12;
+    private final byte[] hKey;
+    private final int securityDescriptorInfo;
+    private final int securityDescriptorSize;
+
+    public BaseRegGetKeySecurityRequest(final byte[] hKey, final int securityDescriptorInfo,
+            final int securityDescriptorSize) {
+        super(OP_NUM);
+        this.hKey = hKey;
+        this.securityDescriptorInfo = securityDescriptorInfo;
+        this.securityDescriptorSize = securityDescriptorSize;
+    }
+
+    @Override
+    public void marshal(final PacketOutput out) throws IOException {
+        out.write(hKey);
+        out.writeInt(securityDescriptorInfo);
+        final RPCSecurityDescriptor sd = new RPCSecurityDescriptor();
+        sd.setCbInSecurityDescriptor(securityDescriptorSize);
+        sd.setSecurityDescriptor(new byte[securityDescriptorSize]);
```

Comment:
I don't think you need to do both here.

You mentioned that Microsoft isn't expecting a padded `byte[]`? If so, then this won't work, and instead you should only use `setCbInSecurityDescriptor`.

If they do expect a padded `byte[]`, then we can simplify this to only use `setSecurityDescriptor`. The `CbInSecurityDescriptor` and `CbOutSecurityDescriptor` are both implied in that case.

Reply:
I do need to set the both since we have to give them a byte array of arbitrary content.